packages/smooth_app/lib/pages/history_page.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/data_models/product_list.dart';
+import 'package:smooth_app/database/dao_product_list.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/pages/product/common/product_list_page.dart';
+
+class HistoryPage extends StatefulWidget {
+  const HistoryPage();
+  @override
+  State<HistoryPage> createState() => _HistoryPageState();
+}
+
+class _HistoryPageState extends State<HistoryPage> {
+  final ProductList productList = ProductList(
+    listType: ProductList.LIST_TYPE_HISTORY,
+    parameters: '',
+  );
+
+  @protected
+  @mustCallSuper
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    DaoProductList(context.watch<LocalDatabase>()).get(productList).then((_) {
```

Comment:
is it cheap to call `context.watch<LocalDatabase>()` multiple times ? 
If yes, then LG
If no, another option is to set it up in `initState`

Reply:
If I understand it correctly, we don't have a `context` in `initState`, so that won't work (https://stackoverflow.com/questions/58371874/what-is-the-difference-between-didchangedependencies-and-initstate).

We could of course use some kind of do-it-once thing, a bool or something else, but I suspect (I don't know - I haven't found very informative docs about this) that `watch` just registers for updates when notify is called.

Reply:
yeah, I can't find it now, but I read once that watch is just O(1) operation. 

Go for it.

Reply:
Checks are failing though. I'll take a look tomorrow.

Reply:
Those checks are failing for everyone :( 

I have to look  into it deeper, couldn't find the reason on the surface, but I think you can submit this for now.

Reply:
Already fixed by https://github.com/openfoodfacts/smooth-app/pull/656, merged it a few minutes ago